src/transport/SecureSessionMgr.cpp

```diff
@@ -341,6 +345,44 @@ void SecureSessionMgr::SecureMessageDispatch(const PacketHeader & packetHeader,
         ExitNow(err = CHIP_ERROR_KEY_NOT_FOUND_FROM_PEER);
     }
 
+    if (packetHeader.GetFlags().Has(Header::FlagValues::kSecure))
```

Comment:
This check does not make sense to me.  If we got here, that flag is set, no?  The only call into this function is guarded by this flag.....